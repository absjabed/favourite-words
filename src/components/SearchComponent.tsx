import React from 'react';
import {StyleSheet, View} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5'
import Colors from '../constants/Colors/Colors';
import Row from '../screens/Layout/Row';
import AppButton from './Button';
import AppTextInput from './TextInput';


type Props = {
    initiateSearch: () => void;
    onSearchWordChange: (word: string) => void;
    searchedWord: string;
  };

const SearchComponent = ({initiateSearch, onSearchWordChange, searchedWord}: Props) => {
  return (
    <Row style={styles.container}>
        <View style={{width:'80%'}}>
            <AppTextInput
                placeholder="Search Dictionary"
                value={searchedWord}
                onChangeText={(value: string)=> onSearchWordChange(value)}
                showSoftInputOnFocus={false}
            />
        </View>
        <View style={styles.buttonStyle}>
            <AppButton onPress={()=>  initiateSearch()} title="Search"></AppButton>
        </View>
        <Icon style={styles.iconStyle} name="search" size={20} color={Colors.lightGray} />
    </Row>
  );
};

export default SearchComponent;

const styles = StyleSheet.create({
  container: {
    display:'flex', 
    flexDirection:'row', 
    width:'100%', 
    justifyContent:'center', 
    alignContent:'center', 
    alignItems:'center'
  },
  buttonStyle:{
      paddingBottom:10, 
      justifyContent:'center', 
      alignItems:'center'
    },
  iconStyle: {
      position:'absolute', 
      left:10, 
      bottom:15
    },
});
