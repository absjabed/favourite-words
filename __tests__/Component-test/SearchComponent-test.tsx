import 'react-native';
import React from 'react';
import SearchComponent from '../../src/components/composite/SearchComponent';

import renderer, { create } from 'react-test-renderer';

const mockSearchedWord = "Owl",
  mockinitiateSearch = () => console.log("search initiated"),
  mockOnSearchWordChange = () => console.log("search word changed");

it('SearchComponent renders correctly', () => {
  renderer.create(<SearchComponent initiateSearch={mockinitiateSearch} searchedWord={mockSearchedWord} onSearchWordChange={mockOnSearchWordChange} />);
});

test('SearchComponent snapshot', () => {
  const tree = create(<SearchComponent initiateSearch={mockinitiateSearch} searchedWord={mockSearchedWord} onSearchWordChange={mockOnSearchWordChange} />);;
  expect(tree).toMatchSnapshot();
});
