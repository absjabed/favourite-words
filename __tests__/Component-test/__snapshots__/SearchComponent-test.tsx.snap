// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "width": "100%",
      },
      Object {
        "alignContent": "center",
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginLeft": 25,
        "paddingLeft": 3,
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "borderColor": "#5f5f5f",
          "borderRadius": 7,
          "borderWidth": 1,
          "height": 40,
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 10,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        blurOnSubmit={false}
        onChangeText={[Function]}
        placeholder="Search Dictionary"
        placeholderTextColor="#9b9b9b"
        rejectResponderTermination={true}
        showSoftInputOnFocus={false}
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Ubuntu-Regular",
            "opacity": 100,
            "paddingHorizontal": 15,
            "textAlign": "auto",
          }
        }
        textAlign="left"
        underlineColorAndroid="transparent"
        value="Owl"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingBottom": 10,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#555CC4",
          "borderRadius": 7,
          "color": "#ffffff",
          "marginLeft": 20,
          "marginRight": 20,
          "marginTop": 16,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#ffffff",
            "fontFamily": "Ubuntu-Regular",
          }
        }
      >
        Search
      </Text>
    </View>
  </View>
  <Icon
    color="#9b9b9b"
    name="search"
    size={20}
    style={
      Object {
        "bottom": 15,
        "position": "absolute",
        "right": 135,
      }
    }
  />
</View>
`;
